### ISSUE TITLE
castArray error

### ISSUE THREAD

**Issue by librehome** [TypeScriptToLua/TypeScriptToLua#985]
The castArray won't compile against later versions (e.g. 4.1.3) of TypeScript.
I made the changes in utils.ts:

```
export function castArray<T>(value: T | readonly T[]): readonly T[] {
    if (Array.isArray(value)) {
        return value;
    } else {
        const v: T = value as any;
        return [v];
    }
}
```

**Comment by Perryvw** [comment]
This is a TypeScript bug: https://github.com/microsoft/TypeScript/issues/41984 We will not be changing castArray as it is correct. Maybe we can set an upper limit to the TypeScript version. Seems like TS is planning to deliver their fix in 4.2.0, highest working version should be 4.1.2

**Comment by librehome** [comment]
Yes, I agree it is a TS bug. I reported because the playground is using TS 4.1.3 while I couldn't compile.
Let's wait for the fix.
